### DISCUSSION TITLE
Make SceneManager SceneProcessorBase public

### DISCUSSION THREAD

**Discussion by JonasSchnettker** [FirstGearGames/FishNet#219] [Feature Request]
In my case I had to make the SceneProcessorBase public, because I use a loading screen animation. This is onl occuring on the client side. Since SceneProcessorBase is inheriting from Monobehaviour I cannot check IsClient in there. Therefore I need to declare a different SceneProcessorBase via another script, depending on IsClient or IsServer.

**Comment by TrustNoOneElse** [comment]
InstanceFinder.IsClient is not working for you?

**Reply by JonasSchnettker** [reply]
Hm, in this case I have tried it it seems to work. Still question the design choice a bit. In my case, I only need a custom DefaultSceneProcessor when I have a client. In case of the server, I do not want to override anything at all.

**Comment by FirstGearGames** [comment]
I can absolutely pass in the NetworkManager to the processor.

**Reply by JonasSchnettker** [reply]
That is no issue. I might phrased my problem badly.

The only issue is that I use a different sceneProcessor for Server and Client. Correspondent Code:

In SceneManager Line 101-102
`[SerializeField]
 private SceneProcessorBase _sceneProcessor;`
        
As you can see the _sceneProcessor field is private. In my case I want to use the defaultProcessor in the server case and a customProcessor in client case. I could write a workaround here by using different gameObjects for server and client, but this could cause mistakes later. It is far safer to make the field public and specify the _sceneProcessor in code.

**Reply by FirstGearGames** [reply]
So you want it exposed so you can set the processor at runtime? If so, I can definitely do that as well.

**Reply by FirstGearGames** [reply]
Assuming I understood right, this was added in 3.3.0. 
- Added SceneManager.GetSceneProcessor.
- Added SceneManager.SetSceneProcessor.

**Reply by JonasSchnettker** [reply]
Thank you, will use that in the future!
